### PR TITLE
barrier: fix build with gcc

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Fix the build on systems using gcc

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Review by reneeotten — COMMENTED** [review]
did you submit any of these patches upstream? Please add a link or do so

**Comment by reneeotten** [comment]
most of the patches seem like they would be okay regardless if the compiler used and could/should thus be send upstream.

**Comment by barracuda156** [comment]
@reneeotten Ok, I recalled now, I did not submit `cstdint` fix because there is already an existing PR for it: https://github.com/debauchee/barrier/pull/1886

I think everything else is just tweaking the build to work for gcc, which does not support Apple ObjC/ObjC++ syntax fully.
I can try opening PR for some of that, but do not really expect it to get merged.

**Comment by barracuda156** [comment]
@reneeotten @herbygillot Could we merge this?

As you can see from the fact that an absolutely credible and minimal fix for modern gcc is still pending in upstream, they are not quick with merging anything, it is pointless to wait. There were folks wanting specifically this port on powerpc, let them have it, since I doubt they are using my fork of MacPorts, and also they were on Leopard, not 10.6 ppc.

File: aqua/barrier/Portfile [hunk 1]
```diff
@@ -46,15 +44,33 @@ if { ${os.major} < 16 } {
     configure.args-append \
```
Comment (reneeotten) [inline aqua/barrier/Portfile:44]: this needs the `os.platform` check as well

File: aqua/barrier/Portfile [hunk 2]
```diff
@@ -46,15 +44,33 @@ if { ${os.major} < 16 } {
...
         -DBARRIER_BUILD_GUI=OFF
 } else {
-    PortGroup  qt5 1.0
-    PortGroup           app 1.0
+    PortGroup   qt5 1.0
+    PortGroup   app 1.0
 
     qt5.depends_component \
         qtbase
```
Comment (reneeotten) [inline aqua/barrier/Portfile:51]: this should be implied by the `qt5` PortGroup and as you see `port lint` throws a warning about it; please remove this and check

Comment (barracuda156) [reply aqua/barrier/Portfile]: @reneeotten Removed that line.

File: aqua/barrier/Portfile (hunk 1 above)
Comment (reneeotten) [reply aqua/barrier/Portfile:44]: please make this last requested change and we can merge

Comment (barracuda156) [reply aqua/barrier/Portfile:44]: Done

**Pull request merged**